Source/MQTTnet/Implementations/MqttTcpChannel.cs

```diff
@@ -127,6 +127,11 @@ public async Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancellat
                 // Workaround for: https://github.com/dotnet/corefx/issues/24430
                 using (cancellationToken.Register(Dispose))
                 {
+                    if (this._stream == null)
```

Comment:
fetch the field into a local variable and perfom the check and the ReadAsync on the local variable otherwise it is not threadsafe

Reply:
Makes sense 👍

Reply:
Is resolved now.